Core/include/Acts/TrackFinding/CombinatorialKalmanFilter.hpp

```diff
@@ -394,10 +371,42 @@ class CombinatorialKalmanFilter {
         }
       }
 
-      // Reset propagation state:
-      // - When navigation breaks and there is still active tip present after
-      // recording&removing track tips on current surface
-      if (navigator.navigationBreak(state.navigation) && !result.filtered) {
+      bool isEndOfWorldReached =
+          endOfWorldReached(state, stepper, navigator, logger());
+      bool isPathLimitReached =
+          result.pathLimitReached(state, stepper, navigator, logger());
+      bool isTargetReached = targetReached(state, stepper, navigator, logger());
+      bool isNavigationBreak = navigator.navigationBreak(state.navigation);
```

Comment:
```suggestion
      const bool isEndOfWorldReached =
          endOfWorldReached(state, stepper, navigator, logger());
      const bool isPathLimitReached =
          result.pathLimitReached(state, stepper, navigator, logger());
      const bool isTargetReached = targetReached(state, stepper, navigator, logger());
      const bool isNavigationBreak = navigator.navigationBreak(state.navigation);
```
I was double checking, what we are doing with those values. Maybe `const` makes it clearer, that we only need them for the next large scope?

Reply:
sure